packages/widget/src/components/SendToWallet/SendToWallet.tsx

```diff
@@ -38,44 +50,52 @@ export const SendToWallet: React.FC<BoxProps> = forwardRef((props, ref) => {
         !hiddenToAddress,
     ) || requiredToAddress;
 
-  const {
-    field: { onChange, onBlur, name, value },
-  } = useController({
-    name: FormKey.ToAddress,
-    rules: {
-      required:
-        requiredToAddress && (t('error.title.walletAddressRequired') as string),
-      onChange: (e) => {
-        setValue(FormKey.ToAddress, e.target.value.trim());
-      },
-      validate: async (value: string) => {
-        try {
-          if (!value || isAddress(value) || isSVMAddress(value)) {
-            return true;
-          }
-          const address = await getEnsAddress(config, {
-            chainId: getValues(FormKey.ToChain),
-            name: normalize(value),
-          });
-          return Boolean(address);
-        } catch (error) {
-          return t('error.title.walletEnsAddressInvalid') as string;
+  useEffect(() => {
+    addFieldValidation('toAddress', async (value: string) => {
+      try {
+        if (!value && requiredToAddress) {
+          return t('error.title.walletEnsAddressInvalid');
         }
-      },
-      onBlur: () => trigger(FormKey.ToAddress),
-    },
-  });
+
+        if (!value || isAddress(value) || isSVMAddress(value)) {
+          return true;
+        }
+        const address = await getEnsAddress(config, {
+          chainId: getFieldValues('toChain')[0],
+          name: normalize(value),
+        });
+        return Boolean(address);
+      } catch (error) {
+        return t('error.title.walletEnsAddressInvalid') as string;
+      }
+    });
+  }, [addFieldValidation, requiredToAddress, getFieldValues, t, config]);
```

Comment:
Should we add this field onBlur instead of creating validation on every field update?

Reply:
We discussed this and decide for now its actually ok for it to stay here as it only really will update on requiredToAddress and config changing - though we might look to optimise the pattern for validation in future.

When testing this is wasn't actually working. This was main due to a problem in the useValidationActions hook and a missing call to clearErrors when valid. 

Now fixed and works as it did previously  - the additional clearErrors logic has been put in the formStore logic and the useValidationActions has been rewritten like the useFieldActions hook so that it avoid any potential problems related to connascence of position/meaning. See commit https://github.com/lifinance/widget/pull/166/commits/9ebef7fd5b78e219d1aeb961b070ac654fc47c14